src/processor/operator/persistent/node_batch_insert.cpp

```diff
@@ -124,8 +126,20 @@ void NodeBatchInsert::executeInternal(ExecutionContext* context) {
         token = nodeLocalState->localIndexBuilder->getProducerToken();
     }
 
+    auto nodeInfo = ku_dynamic_cast<BatchInsertInfo*, NodeBatchInsertInfo*>(info.get());
     while (children[0]->getNextTuple(context)) {
         auto originalSelVector = nodeLocalState->columnState->getSelVectorShared();
+        auto numTuples = originalSelVector->getSelSize();
```

Comment:
I'd separate this into a function, which can: 1. better isolate the operation; 2. give it a proper name so easier to understand the flow here inside while loop.